tests/test_parser.py

```diff
@@ -1021,3 +1024,89 @@ def test_bthome_moisture(caplog):
             ),
         },
     )
+
+
+def test_bthome_event_switch_single_press(caplog):
+    """Test BTHome parser for an event of a single press of a switch without encryption."""
+    data_string = b"\x02\x3B\x05"
+    advertisement = bytes_to_service_info(
+        data_string, local_name="TEST DEVICE", address="A4:C1:38:8D:18:B2"
+    )
+
+    device = BTHomeBluetoothDeviceData()
+
+    assert device.update(advertisement) == SensorUpdate(
+        title="TEST DEVICE 18B2",
+        devices={
+            None: SensorDeviceInfo(
+                name="TEST DEVICE 18B2",
+                manufacturer=None,
+                model="BTHome sensor",
+                sw_version="BTHome BLE",
+                hw_version=None,
+            )
+        },
+        entity_descriptions={
+            KEY_SIGNAL_STRENGTH: SensorDescription(
+                device_key=KEY_SIGNAL_STRENGTH,
+                device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+                native_unit_of_measurement=Units.SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+            ),
+        },
+        entity_values={
+            KEY_SIGNAL_STRENGTH: SensorValue(
+                device_key=KEY_SIGNAL_STRENGTH, name="Signal Strength", native_value=-60
+            ),
+        },
+        events={
+            KEY_SWITCH: Event(
+                device_key=KEY_SWITCH,
+                name="Switch",
+                event_type="single_press",
+                event_subtype=None,
+            ),
+        },
+    )
+
+
+def test_bthome_event_dimmer_rotate_left_3_steps(caplog):
+    """Test BTHome parser for an event rotating a dimmer 3 steps left without encryption."""
+    data_string = b"\x03\x3C\x0C\x03"
+    advertisement = bytes_to_service_info(
+        data_string, local_name="TEST DEVICE", address="A4:C1:38:8D:18:B2"
+    )
+
+    device = BTHomeBluetoothDeviceData()
+
+    assert device.update(advertisement) == SensorUpdate(
+        title="TEST DEVICE 18B2",
+        devices={
+            None: SensorDeviceInfo(
+                name="TEST DEVICE 18B2",
+                manufacturer=None,
+                model="BTHome sensor",
+                sw_version="BTHome BLE",
+                hw_version=None,
+            )
+        },
+        entity_descriptions={
+            KEY_SIGNAL_STRENGTH: SensorDescription(
+                device_key=KEY_SIGNAL_STRENGTH,
+                device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+                native_unit_of_measurement=Units.SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+            ),
+        },
+        entity_values={
+            KEY_SIGNAL_STRENGTH: SensorValue(
+                device_key=KEY_SIGNAL_STRENGTH, name="Signal Strength", native_value=-60
+            ),
+        },
+        events={
+            KEY_DIMMER: Event(
+                device_key=KEY_DIMMER,
+                name="Dimmer",
+                event_type="rotate_left",
+                event_subtype="3",
```

Comment:
feels a bit of a hack to make this a string? It also doesn't feel like a "subtype"

Reply:
I was told to use strings here. https://github.com/Bluetooth-Devices/sensor-state-data/pull/38#discussion_r984107132

I can also combine it in event_type as “rotate_left_3” or “rorate_left_3_steps”, is that what is preferred?

Reply:
I am not sure if @bdraco realized that it would store an integer. If we can parse it, why wouldn't we allow more complicated parsed data ? An `event_data` key containing a dict that we will merge into the event data in HA.

Reply:
I think it is fine to add `int` there.  I was mostly concerned about the broad typing there i.e. `date` or `datetime` seems like it doesn't make much sense as a subtype unless I'm missing something

Reply:
Yeah, datetime is probably not something that will be used quickly. 

How about @balloob ‘s suggestion using a dict `event_data`? Is that better than the “event_type” + “event_subtype”. I guess it is more flexible for other integration, as each integration can define its own “event_data” format.

Reply:
Seems fine to me.

Reply:
Ok, I'll update sensor-state-data first than.

Reply:
We can always update sensor-state-data later to add `date` and/or `datetime` in the dict.